### PR TITLE
upgrade guzzlehttp/guzzle to ^7.2.0 and php to ^7.2

### PR DESCRIPTION
**Pull request by rollo-smartframe**

### CODE REVIEW

File: src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php [hunk 1]
```diff
@@ -14,6 +14,6 @@ function it_is_initializable()
 
     function it_returns_a_psr7_response()
     {
-        $this->send('POST', 'http://endpoint.com/some/uri')->shouldHaveType('Psr\Http\Message\ResponseInterface');
+        $this->send('POST', 'http://smartframe.io')->shouldHaveType('Psr\Http\Message\ResponseInterface');
```
Comment (Integralist) [inline src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php:17]: We would prefer to not reference your own company in these files, but instead opt for something either generic (like `https://httpbin.org`) or keep it set to what it is currently.

**Review by Integralist — CHANGES_REQUESTED** [review]
Hi @rollo-smartframe thanks for opening this PR. 

I have just one requested amendment. 

Thanks.

File: src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php (hunk 1 above)
Comment (rollo-smartframe) [reply src/spec/Fastly/Adapter/Guzzle/GuzzleAdapterSpec.php:17]: Hi, sure previous link did not work, I will change it to https://httpbin.org as you suggest.

**Review by Integralist — APPROVED** [review]
This looks good to me. Thank you for the update.

**Comment by rollo-smartframe** [comment]
@Integralist hi could you also do a release/tag of version 0.4?

**Comment by Integralist** [comment]
@rollo-smartframe of course!
https://github.com/fastly/fastly-php/releases/tag/v0.4.0

**Comment by rollo-smartframe** [comment]
> @rollo-smartframe of course!
> https://github.com/fastly/fastly-php/releases/tag/v0.4.0

Could you also make it available on packagist; there https://packagist.org/packages/fastly/fastly please

**Comment by Integralist** [comment]
@rollo-smartframe that's done now too, thanks! In future new releases will be auto-published to Packagist.

**Pull request merged**